File: 1406-subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.java
class Solution {
    public int subtractProductAndSum(int n) {
     int product =1;
     int sum = 0;
     int result = 0;
     String s = String.valueOf(n);
     int [] ar = new int[s.length()];
     for(int i=0;i<s.length();i++){
         ar[i] = s.charAt(i) - '0';
         sum=sum+ar[i];
         product=product*ar[i];
         result = product - sum;
      }
      return result;
    }
}

File: 1406-subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function(n) {
 
    let ar = n.toString().split('');
    let product = 1;
    let sum = 0;
    let output =0;

    for(let i = 0; i < ar.length; i++) {
       let num = Number(ar[i]);
        product *=num;
        sum +=num;
        output = product - sum;
    }

    return output;
};